src/firefly/js/visualize/ui/MatchLockDropDown.jsx

```diff
@@ -6,14 +6,41 @@ import {PlotAttribute} from '../PlotAttribute';
 import {SingleColumnMenu} from '../../ui/DropDownMenu.jsx';
 import {DropDownVerticalSeparator, ToolbarButton} from '../../ui/ToolbarButton.jsx';
 import {DropDownToolbarButton} from '../../ui/DropDownToolbarButton.jsx';
-
 import MATCH_LOCKED from 'html/images/28x28_Match_Locked.png';
 import MATCH_UNLOCKED from 'html/images/28x28_Match_Unlocked.png';
 import {isImage} from '../WebPlot';
+import {showInfoPopup} from '../../ui/PopupUtil';
 
 function changeMatchType(vr, matchType, lockMatch) {
     const plot= primePlot(vr);
     if (!plot) return;
+    //check for images without WCS when aligning by WCS or Target, and display appropriate warning
+    if(matchType.key === 'Standard' || matchType.key === 'Target') {
+        const warningTitles = [];
+        getPlotViewAry(vr)?.forEach( (currPlots) => {
+            currPlots.plots?.forEach( (plt) => {
```

Comment:
You don't have to loop though all the plots in a PlotView. So the nested loop is unnecessary. Only the `primePlot` matters in the context. Also our idiom for the `PlotView` is `pv`. this this could be 

```js
getPlotViewAry(vr)?.forEach( (pv) => {
      const p= primePlot(pv);
```

Reply:
I really think filter/map is a better approach than forEach

Something like....
```js
const warningTitles= getPlotViewAry(vr)?
                .filter( (pv) => hasWCSProjection(primePlot(pv));
                .map ( (pv) => primePlot(pv).title
```

Reply:
Done. The test built is updated too. Let me know if it looks okay @robyww.